### ISSUE TITLE
2-way setting

### ISSUE THREAD

**Issue by roywithfiringblade** [Xiangyi1996/PPNet-PyTorch#6]
Thanks for sharing your work! Could you share the experiment setting of the 2 -way result in the article? The Meta SegNet has some special design for PASCAL, and they only evaluate the performance on fold 2, which contains the person class. And what is the number of query images per task. The default setting in PANet is 1, which is not reasonable in my opinion. Cause it might not contain all 2 classes.

**Comment by Xiangyi1996** [comment]
The number of query images per task is 1. **2-way K-shot ** setting means there are 2 classes in support set and K support images per class. And there is 1 query image which has object categorys belongs to these 2 classes.

**Comment by roywithfiringblade** [comment]
@Xiangyi1996 Thanks for reply, so is the performance evaluated on just fold 2 like Meta SegNet or on all 4 folds?

**Comment by Xiangyi1996** [comment]
Our performances are all evaluated on 4 folds. You can see in Table 1,2,3,4

**Comment by roywithfiringblade** [comment]
@Xiangyi1996 I just check the images in PASCAL, the 1449 val images have only a few with 2 classes in one images. For example, the fold1 contains 22 and fold 0 only contains 1 images. My dataset is the same as CANet, could you share the detailed setting, Thanks!

**Comment by youngfly11** [comment]
I believe that you may have some misunderstanding of the test set in few-shot segmentation. For example, if we have 20 categories in pascal, we split the first 15 classes as training classes, which contains all images (in the original pascal training set, validation set, test set) to construct the few-shot training set.  So the few-shot test set are constructed by all images, which contains the left 5 classes.

**Comment by roywithfiringblade** [comment]
@youngfly11 As the Chapter 5 in OSLSM mentioned, The training set is formed by inclunding all image-mask pairs from PASCAL VOC training set, and the test set image-label pairs are taken from PASCAL VOC validation set. Not sure if i understand it correctly.

**Comment by Xiangyi1996** [comment]
@704051419 As I mentioned before_, **2-way K-shot ** setting means there are 2 classes in support set and K support images per class. And there is 1 query image which has object categorys belongs to these 2 classes._ It may contain one class from these two categories or contain all these two categories. So the 2 classses images number you mentioned is not a limitation in sampling images.
If you have more questions, please refer to our dataloader part in constructing tasks. Thanks.
